Automattic/woocommerce-payments
# Adds Stripe metadata to UPE transactions

Fixes #2701 

#### Changes proposed in this Pull Request


# Adding Metadata to Stripe Transactions

Currently, transactions made using the UPE are not sending metadata values to Stripe prior to confirming the intention. The fields included in the metadata are `customer_name`, `customer_email`, `site_url`, `order_id`, `order_key`, and `payment_type`. You can confirm by visiting the Stripe dashboard for your site and viewing payments made with the UPE. If you select a payment and then scroll down to the "Metadata" section, you will be disappointed to see that "No metadata" is displayed.

![Missing metadata](https://cdn-std.droplr.net/files/acc_1104206/tEmE1Y)
_Missing metadata in Stripe dashboard_

The parent payment gateway sends metadata values when confirming a payment intention and here is an example of what this section should normally look like.

![Metadata present](https://cdn-std.droplr.net/files/acc_1104206/KmS1Va)
_Example expected payment metadata_

This PR simply adds these corresponding metadata values for an order when necessary, which in this case is when we update the payment intention before UPE confirmation. 







#### Testing instructions

1. Please pull the changes in this PR, the latest version of the Payments Server, and enable UPE checkouts and additional payment methods. 
2. Add an item to your cart and complete a successful checkout.
3. Note the order number, now visit your site's [Stripe dashboard](https://dashboard.stripe.com), find the corresponding payment for the completed order, scroll down, and check the metadata values. 
4. If the metadata values match what has been saved to the order, rejoice. 






-------------------
